torch/fx/passes/utils/matcher_utils.py

```diff
@@ -204,41 +205,35 @@ def _match_nodes(self, pn: Node, gn: Node, match: InternalMatch) -> bool:
         # Recursively traverse upwards to check if `pn` is a true
         # match for `gn`
         match_found = True
-
-        def flatten_args(args) -> List[Any]:
-            # Recursively flatten args
-            result : List[Any] = []
-            for arg in args:
-                # flatten the list, if only it's a list/tuple of nodes
-                if isinstance(arg, (list, tuple)) and len(arg) > 0 and isinstance(arg[0], Node):
-                    result.extend(flatten_args(arg))
+        
+        def _match_args(args1: Union[List, Tuple], args2: Union[List, Tuple]) -> bool:
+            if len(args1) != len(args2):
+                return False 
+
+            for a1, a2 in zip(args1, args2):
+                if isinstance(a1, Node) and isinstance(a2, Node):
+                    matched = self._match_nodes(a1, a2, match)
+                elif isinstance(a1, (list, tuple)) and isinstance(a2, (list, tuple)):
+                    matched = _match_args(a1, a2)
```

Comment:
you can pytree.tree_flatten the args first, so that it become a flat list. 
Then you don't need to handle `isinstance(a1, (list, tuple))` case.

Reply:
But if we have the following graphs:
```
torch.ops.aten._reshape_alias_copy.default(x, [1, y.shape[0]], [y.shape[1], y.shape[1]])
```
and
```
torch.ops.aten._reshape_alias_copy.default(x, [b, y.shape[0], y.shape[1]], [y.shape[1]])
```

Wouldn't they get treated the same after flattening?

Reply:
pytree.tree_flatten also return a TreeSpec, which can be used to disambiguate this case.

Reply:
This doesn't handle the case with comparing a placeholder against a list of literals:
```

Reply:
ic... 
placeholder is a wildcard in this case, and can be match with any python object, including list/tuple. 
Signing off.

Reply:
Maybe add a comment for this case, in case reader forgot about "placeholder can be a wildcard" thing.